proto/query.proto

```diff
@@ -203,6 +239,14 @@ message Field {
 
   // vitess-defined type. Conversion function is in sqltypes package.
   Type type = 2;
+  string table = 3;
```

Comment:
I would add a comment before this group of fields saying they are added only with the query option.